pkg/engine/engine.go

```diff
@@ -468,14 +560,20 @@ func filterDetectors(filterFunc func(detectors.Detector) bool, input []detectors
 	return out
 }
 
-// initialize prepares the engine's internal structures. The LRU cache optimizes
+// initialize prepares the engine's internal structures. The LRU detectionCache optimizes
 // deduplication efforts, allowing the engine to quickly check if a chunk has
 // been processed before, thereby saving computational overhead.
 func (e *Engine) initialize(ctx context.Context) error {
 	// TODO (ahrav): Determine the optimal cache size.
-	const cacheSize = 512 // number of entries in the LRU cache
+	const (
+		cacheName = "dedupe_cache"
+		cacheSize = 512 // number of entries in the LRU cache
```

Comment:
I suspect the optimal size is at least 4x this.

Reply:
I agree, but we can address this in a separate PR to keep the focus of this one on the detection cache.